src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -7273,14 +7273,21 @@ and TcObjectExpr (cenv: cenv) env tpenv (objTy, realObjTy, argopt, binds, extraI
         overridesAndVirts |> List.iter (fun (m, implTy, dispatchSlots, dispatchSlotsKeyed, availPriorOverrides, overrides) ->
             let overrideSpecs = overrides |> List.map fst
             let hasStaticMembers = dispatchSlots |> List.exists (fun reqdSlot -> not reqdSlot.MethodInfo.IsInstance)
+            let isOverallTyAbstract =
+                match tryTcrefOfAppTy g objTy with
+                | ValueNone -> false
+                | ValueSome tcref -> HasFSharpAttribute g g.attrib_AbstractClassAttribute tcref.Attribs
+                
+            if overrideSpecs.IsEmpty && not (isInterfaceTy g objTy) then
+                errorR (Error(FSComp.SR.tcInvalidObjectExpressionSyntaxForm (), mWholeExpr))
```

Comment:
Can you please briefly comment on why is this change here necessary if we only want to rewrite an expression? Is it because we now checking one additional "form" of object expression?

Reply:
@vzarytovskii  See [comment](https://github.com/dotnet/fsharp/pull/17388/files#r1667673941), basically we want to keep the same error reporting.
Added tests to show this